From 17a924a2275b3e7721dc05e0670cae638371a98f Mon Sep 17 00:00:00 2001
From: Jake Dane <3689-jakedane@users.noreply.gitlab.gnome.org>
Date: Fri, 19 Aug 2022 06:12:25 +0000
Subject: [PATCH] jpeg: Increase memory limit for loading image data

As fix for security issue #205 when loading image data the memory size
was limited to 100 MB. That seemed like a good threshold. For larger
images, from around 18 megapixels (MP) and up though not for all such
images, this threshold was too low. Increasing the threshold too 300 MB
seems to work better and lets larger images load.

Fixes #216.
---
 gdk-pixbuf/io-jpeg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk-pixbuf/io-jpeg.c b/gdk-pixbuf/io-jpeg.c
index 22f4174fe..6317e20d3 100644
--- a/gdk-pixbuf/io-jpeg.c
+++ b/gdk-pixbuf/io-jpeg.c
@@ -1091,7 +1091,7 @@ gdk_pixbuf__jpeg_image_load_increment (gpointer data,
 			rc = jpeg_read_header (cinfo, TRUE);
 			context->src_initialized = TRUE;
 
-                        cinfo->mem->max_memory_to_use = 100 * 1024 * 1024;
+                        cinfo->mem->max_memory_to_use = 300 * 1024 * 1024;
 			
 			if (rc == JPEG_SUSPENDED)
 				continue;
-- 
GitLab

